### ISSUE TITLE
x0vncserver stops after first connection

### ISSUE THREAD

**Issue by mrsteve0924** [TigerVNC/tigervnc#1595]
**Describe the bug**
running `x0vncserver rfbauth ~/.vnc/passwd &`  from  ~/.xprofile
i can connect to server once using vncviewer from android. but attempting to connect a second time gives me an error 
"the port on which the computer is listening for a connection could not be contacted"

if i instead start the server from the terminal and leave running i get the same behavior
`x0vncserver rfbauth ~/.vnc/passwd` or
`x0vncserver rfbauth ~/.vnc/passwd -display :0  -rfbport 5900 `
 

**Server (please complete the following information):**
 - OS: archlinux kernel 6.1.9
 - VNC server: TigerVNC
 - VNC server version: 1.13.0-1
 - Server downloaded from: [my Linux distribution]

logs after connection/disconnection form vncviewer.  
```
x0vncserver -rfbauth ~/.vnc/passwd_arch

Mon Feb  6 23:18:04 2023
 Geometry:    Desktop geometry is set to 3840x1080+0+0
 XDesktop:    Using evdev codemap
 XDesktop:    
 XDesktop:    XTest extension present - version 2.2
 XDesktop:    RANDR extension not present
 XDesktop:    Will not be able to handle session resize
 Main:        Listening for VNC connections on all interface(s), port 5900

Mon Feb  6 23:21:25 2023
 Connections: accepted: 192.168.1.101::39078
 SConnection: Client needs protocol version 3.8
 SConnection: Client requests security type VncAuth(2)
 XDesktop:    Enabling 8 buttons of X pointer device
 XDesktop:    Allocated shared memory image
 VNCSConnST:  Server default pixel format depth 24 (32bpp) little-endian rgb888
 VNCSConnST:  Client pixel format depth 6 (8bpp) rgb222
 VNCSConnST:  Client pixel format depth 24 (32bpp) little-endian rgb888

Mon Feb  6 23:22:12 2023
 VNCSConnST:  closing 192.168.1.101::39078: Clean disconnection
 EncodeManager: Framebuffer updates: 163
 EncodeManager:   RRE:
 EncodeManager:     Solid: 60 rects, 1.26544 Mpixels
 EncodeManager:            1.17188 KiB (1:4218.74 ratio)
 EncodeManager:   ZRLE:
 EncodeManager:     Bitmap RLE: 12 rects, 470.978 kpixels
 EncodeManager:                 838 B (1:2248.28 ratio)
 EncodeManager:     Indexed RLE: 220 rects, 4.50468 Mpixels
 EncodeManager:                  148.94 KiB (1:36.5849 ratio)
 EncodeManager:     Full Colour: 532 rects, 14.866 Mpixels
 EncodeManager:                  2.02644 MiB (1:27.9878 ratio)
 EncodeManager:   Total: 824 rects, 21.1071 Mpixels
 EncodeManager:          2.17383 MiB (1:31.5855 ratio)
 Connections: closed: 192.168.1.101::39078
 ComparingUpdateTracker: 129.667 Mpixels in / 15.6123 Mpixels out
 ComparingUpdateTracker: (1:8.30543 ratio)
X Error of failed request:  BadValue (integer parameter out of range for operation)
  Major opcode of failed request:  135 (XKEYBOARD)
  Minor opcode of failed request:  9 (XkbSetMap)
  Value in failed request:  0x11ff0aff
  Serial number of failed request:  2030
  Current serial number in output stream:  2031


```

**Comment by CendioOssman** [comment]
Looks like a bug in the new keyboard handling.

I'm not able to reliably trigger this here, though. What X server are you using with x0vncserver? It doesn't have RANDR, so I'm guessing it's not normal Xorg?

**Comment by mrsteve0924** [comment]
I'm using Xorg.  i also have xorg-xrandr installed

**Comment by CendioOssman** [comment]
That's odd. RANDR should generally always be enabled for Xorg. I don't even know if it's possible to disable it...

Still, I think I was able to find the issue. Should be fixed in 1e3484f2017f038dd5149cd50741feaf39a680e4.

A shame no one tested the beta to find this. We probably need to release a 1.13.1, given how severe this bug is.

We'll wait a few days and see if there are any more bugs that were overlooked.

**Comment by mrsteve0924** [comment]
Hey that's great news!  Thanks

**Comment by mozo78** [comment]
I can confirm the bug ot three machines with Arch Linux.

**Comment by greyhair-atx** [comment]
On RHEL 8 with Gnome opening a x0vncserver session.  The VNC Client connects successfully and can view the desktop. **Once the VNC session is closed the entire x-session is destroyed** (any open files are lost, etc.).

I've installed tigervnc-server-1.13.0-1.el8.x86_64.rpm and tigervnc-1.13.0.1-1.el8.x86_64.rpm on a RHEL 8 VM and again, once the VNC session is closed the x-session is destroyed (any open files are lost, etc.).  

The error highlights are:
> Feb 22 07:39:08 testGUI.examplei.com gsd-power[44424]: The program 'gsd-power' received an X Window System error.
> (Details: serial 173 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-media-keys[44472]: The program 'gsd-media-keys' received an X Window System error.
> (Details: serial 167 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-wacom[44435]: The program 'gsd-wacom' received an X Window System error.
> (Details: serial 167 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com evolution-alarm[44567]: The program 'evolution-alarm-notify' received an X Window System error.
> (Details: serial 152 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-color[44453]: The program 'gsd-color' received an X Window System error.
> (Details: serial 157 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-xsettings[44433]: The program 'gsd-xsettings' received an X Window System error.
> (Details: serial 180 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-clipboard[44460]: The program 'gsd-clipboard' received an X Window System error.
> (Details: serial 190 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com classification-[44565]: The program 'classification-banner' received an X Window System error.
> (Details: serial 514 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gsd-keyboard[44473]: The program 'gsd-keyboard' received an X Window System error.
> (Details: serial 157 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com gnome-terminal-[44889]: The program 'gnome-terminal-server' received an X Window System error.
> (Details: serial 16854 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)
> Feb 22 07:39:08 testGUI.examplei.com vmtoolsd[44581]: The program 'vmware-user' received an X Window System error.
> (Details: serial 2314 error_code 2 request_code 136 (XKEYBOARD) minor_code 8)

**Comment by fengsam6** [comment]
> That's odd. RANDR should generally always be enabled for Xorg. I don't even know if it's possible to disable it...
> 
> Still, I think I was able to find the issue. Should be fixed in [1e3484f](https://github.com/TigerVNC/tigervnc/commit/1e3484f2017f038dd5149cd50741feaf39a680e4).
> 
> A shame no one tested the beta to find this. We probably need to release a 1.13.1, given how severe this bug is.
> 
> We'll wait a few days and see if there are any more bugs that were overlooked.

how can i install tigervnc 1.13.1 in debian 10? I meet similary question,
![image](https://user-images.githubusercontent.com/24472985/233825633-e768d1ff-5834-4e25-97a6-02b94d852f4a.png)
![image](https://user-images.githubusercontent.com/24472985/233825682-8443725a-6ba1-4d56-985a-148c948474ba.png)

my tigervnc-standalone-server/stable,now 1.11.0+dfsg-2+deb11u1 amd64 [installed]

**Comment by CendioOssman** [comment]
This issue only happens with TigerVNC 1.13.0, so that older version will be fine. Whatever issue you are seeing with Pygame has nothing to do with TigerVNC and you'll have to report the issue to them.
